modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,49 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
 
-				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
+				var images = document.querySelectorAll( 'img.wp-image-' + media[ i ].id ),
+					media_sources = media[ i ].media_details.sources;
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
-					var srcset = images[j].getAttribute( 'srcset' );
+
+					if ( ! media_sources || ! media_sources['image/jpeg'] ) {
+						continue;
+					}
+
+					var srcset = images[ j ].getAttribute( 'srcset' ),
+						sizes = media[ i ].media_details.sizes,
+						sizes_keys = Object.keys( sizes ),
+						flag = true;
+
+					// If a full image is present in srcset, it should be updated.
+					if ( srcset && media_sources['image/webp'] ) {
+						srcset = srcset.replace( media_sources['image/webp'].file, media_sources['image/jpeg'].file );
+					}
+
+					for ( var k = 0; k < sizes_keys.length; k++ ) {
+						var media_sizes_sources = media[ i ].media_details.sizes[ sizes_keys[ k ] ].sources;
+						if ( ! media_sizes_sources || ! media_sizes_sources['image/webp'] || ! media_sizes_sources['image/jpeg'] ) {
+							continue;
+						}
+
+						// Check to see if the image src has any size set, then update it.
+						if ( flag && media_sizes_sources['image/webp'].source_url === images[ j ].src ) {
+							images[ j ].src = media_sizes_sources['image/jpeg'].source_url;
+							flag = false;
+						}
```

Comment:
I don't think we really need the `flag` variable at all. We can simply check if the URL per size is the same as `src` and if so replace it. Once we have replaced it, it'll never be the same like any other URL to go through anyway, and that's a cheap enough comparison to make.

Also see my above reply to @adamsilverstein's comment: We should break here if there's no `srcset` attribute to make the loop more efficient, and I think it would be most straightforward to handle `src` in a variable as well.

Something like this:
```suggestion
						if ( media_sizes_sources['image/webp'].source_url === src ) {
							src = media_sizes_sources['image/jpeg'].source_url;
							// If there is no srcset and the src has been replaced, there is nothing more to replace.
							if ( ! srcset ) {
								break;
							}
						}
```

Reply:
Thanks for the review and guidance. It makes sense to make these changes.